### PR TITLE
introduce SigningAccount struct

### PR DESCRIPTION
**Pull request by vogel**
Get rid of makeAuthOnlyAccount and introduce special struct that only has signing data (weight, thresholds, signers).

Signed-off-by: Rafał Malinowski <rafal.przemyslaw.malinowski@gmail.com>

### CODE REVIEW

File: src/transactions/TransactionFrame.cpp [hunk 1]
```diff
@@ -138,17 +139,29 @@ TransactionFrame::addSignature(DecoratedSignature const& signature)
...
-                   account.getAccount().thresholds[0]));
-    signers.insert(signers.end(), account.getAccount().signers.begin(),
-                   account.getAccount().signers.end());
+            Signer(KeyUtils::convertKey<SignerKey>(signingAccount.accountID),
+                   signingAccount.weight));
+    signers.insert(signers.end(), std::begin(signingAccount.signers),
+                   std::end(signingAccount.signers));
 
-    return signatureChecker.checkSignature(account.getID(), signers,
+    auto neededWeight = [&](){;
```
Comment (MonsieurNicolas) [inline src/transactions/TransactionFrame.cpp]: please use a normal assignment instead of lambda

File: src/signature/SigningAccount.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "signature/SigningAccount.h"
+#include "ledger/AccountFrame.h"
+
+namespace stellar
+{
+
+SigningAccount::SigningAccount(AccountID accountID) :
```
Comment (MonsieurNicolas) [inline src/signature/SigningAccount.cpp]: I prefer as a pattern to pass in a const ref and use the copy constructor (consistent with the rest of the code base), similar to what you're doing with the other constructor `SigningAccount(AccountFrame const& accountFrame)`

File: src/signature/SigningAccount.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+#include "xdr/Stellar-ledger.h"
+
+namespace stellar
+{
+
+class AccountFrame;
+
+struct SigningAccount
+{
+    AccountID accountID;
```
Comment (MonsieurNicolas) [inline src/signature/SigningAccount.h]: our code base follows the convention that members are named starting with `m` (lower case m)

File: src/signature/SigningAccount.h [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    xdr::xvector<Signer, 20> signers {};
```
Comment (MonsieurNicolas) [inline src/signature/SigningAccount.h]: do not specify `20` here, it's already taken care of by other objects

Comment (vogel) [reply src/signature/SigningAccount.h]: I I remove 20 form here, I'll get a compilation error in assignment from accountFrame.getAccount().signers

**Comment by MonsieurNicolas** [comment]
I am closing this for now, the naming doesn't match the semantics. We may revive something like this after the datalayer overhaul

**Pull request closed without merging**